sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1647,9 +1647,19 @@ static void ReverseRangeDimensionsForKernel(NDRDescT &NDR) {
 
 pi_result ExecCGCommand::SetKernelParamsAndLaunch(
     CGExecKernel *ExecKernel, RT::PiKernel Kernel, NDRDescT &NDRDesc,
-    std::vector<RT::PiEvent> &RawEvents, RT::PiEvent &Event) {
+    std::vector<RT::PiEvent> &RawEvents, RT::PiEvent &Event,
+    ProgramManager::KernelArgMask EliminatedArgMask) {
+  assert(EliminatedArgMask.empty() ||
+         EliminatedArgMask.size() == ExecKernel->MArgs.size());
+  vector_class<ArgDesc> &Args = ExecKernel->MArgs;
+  std::sort(Args.begin(), Args.end(), [](const ArgDesc &A, const ArgDesc &B) {
```

Comment:
it seems not practical to do this at every launch. Please add a TODO if agreed, and address soon